packages/x-charts/src/LineChart/AreaPlot.tsx

```diff
@@ -97,6 +98,10 @@ const useAggregatedData = () => {
           .x((d) => xScale(d.x))
           .defined((_, i) => connectNulls || data[i] != null)
           .y0((d) => {
+            if (typeof baseline === 'number') {
+              return yScale(baseline)!;
+            }
+
```

Comment:
This only impacts the plotting. Here is an example with values fluctuating between 200-300, and with baseline set to 1000.

Notice that:
- The basline has no impact on the max value of the y-axis
- The yèaxis always start at 0 because it's the assumed baseline

![image](https://github.com/user-attachments/assets/b2e737e3-6182-437e-8556-6c02c0a4b1a8)

Reply:
Potential solution

1. When computing the extremum values, you replace when needed the base (`d[0]`) by the user defined baseline

2. Instead of providing a number, user can provide `'top' | 'bottom'`. Then you just need to ignore the `d[0]` by doing
  ```diff
  - isArea && axis.scaleType !== 'log' ? ...
  + isArea && axis.scaleType !== 'log' && series[seriesId].baseline === undefined ? ...
  ```

  and in the commented line return 
  ```diff
  - return yScale(baseline)!;
  - return baseline==='top' ? yScale.range[0] : yScale.range[1];
  ```

https://github.com/mui/mui-x/blob/38184fe912eb201d29c8e31b8b11f0f0fd854c51/packages/x-charts/src/LineChart/extremums.ts#L43-L44

Reply:
It behaves a bit weirdly when `baseline` changes the extremums.

This happens because the extremum getter value is then passed into the `nice()` function, it can have unintended consequences, which means I would also have to check the baseline where that is calculated.

But we already have the axis `min/max` values for that. So I believe we should use the min/max values when the axis extremums need to be changed.

![screencapture-localhost-3001-playground-area-chart-baseline-2024-08-12-11_20_10](https://github.com/user-attachments/assets/82c80f06-ddff-4821-889b-0556554a41aa)

Reply:
> It behaves a bit weirdly when baseline changes the extremums.

TLTR: I don't think it's an issue

--- 

Technically speaking, we have two behavior to define:
- where does the area set the baseline
- should the basline impact the extremums.

## Should the baseline impact the extremums

For stacked data, I can not think of a usecase where the answer would be no. Otherwise you would compare some area wherase the base one would have missing part.

![image](https://github.com/user-attachments/assets/3130bebc-0fed-41bf-971f-5e40b3bbaf0b)

For a single series, where the area is more aesthetic, seems it can be ignored. Here foe example the NASDAQ evolution during the day. Nobody care of the space between 0 and 16.500. Only the relative evolution matter. Otherwise you would see nothing.

![image](https://github.com/user-attachments/assets/9785ee3b-d1d7-4eaf-8468-673d2840e183)

## Where does the area set the baseline

I only see two reasons to modify the default 0:

- aesthetic reasons. But in that case, only min/max makes sense
- There is a specific value to compare to. For example the average over time. But in that case, I don't see a reason to pick a value that is always on top or below the line


So basically the weird case you define should not exist

![image](https://github.com/user-attachments/assets/6382fe43-3e16-421a-af19-af77cdd18f6d)

Reply:
Ah, but those are just working as expected. The issue that appears when we have the `baseline` affect the extremums is that the extremums will get `niced` after. So if we have a non-round number like below, it will not work properly. Then this starts to get more complex and would clash with a feature that already exists `yAxis.min/max`

![Screenshot 2024-08-12 at 13 03 52](https://github.com/user-attachments/assets/3443f54d-c4ce-40de-817f-fb6276252d27)